changelog/unreleased/enhancement-permission-based-upload-create-buttons

```diff
@@ -0,0 +1,6 @@
+Enhancement: Permission-based upload and create buttons
+
+Instead of showing disabled "Upload" and "New" buttons, we now hide them for the user if they lack permissions for the currently viewed folder.
```

Comment:
The described behaviour is only the desired behaviour for public links. For shares in an authenticated context we ( @tbsbdr and myself ) figured that disabling if the user lacks create permissions is better to avoid a jumpy layout when going back and forth.